schemas/charts/stat/stat.cue

```diff
@@ -15,29 +15,25 @@ package stat
 
 import (
 	"github.com/perses/perses/schemas/common"
-	"github.com/perses/perses/schemas/common/prometheus"
 )
 
-#sparkline: {
-	line_color?:   string
-	line_width?:   number
-	line_opacity?: number
-	area_color?:   string
-	area_opacity?: number
-}
-
 #panel: {
 	kind: "StatChart"
-	display: {
```

Comment:
yes actually you could let this part in stat.cue, it's just that since it defines nothing new compared to the base def, it's useless

Actually I forgot to explain it in the doc, I will add it: the goal was to simplify the creation of plugins for contributors: if you dont add any constraint/new field for a given part of the schema (i.e for now it's mostly about the `display` and `datasource` fields), you can simply remove it, this way you really what this plugin brings in term of new constraints & attributes.

Reply:
hey @eunicorn thanks for replying :).

So just to clarify a bit, @AntoineThebaud made some changes regarding the `#display`. Basically as it is something common to all chart, we decided to put it in the base cue model which is hardcoded in Go.

So my question was more how do you feel to change the base chart cue file in the Go code instead in a cue file ?
Do you think it will be a bit weird to not have the full definition of each panel in the cue file, even if you have to duplicate some common field such as the field `#display`.

It's totally ok if you don't have any opinion on this topic. I was just asking this because you are the first person that is changing these files, so it will impact you directly as soon as this PR is merged.

I hope it clarifies why I pinged you in this conversation :).

Reply:
I think the definition should be in cue regardless of if it is common across all panels, having to touch the Go code makes it less accessible / more overhead when adding properties. I kind of thought this was one of the main benefits of why we decided to use cue?

Reply:
Since it is a based schema that is mandatory and shouldn't modified by the users, it make sense to have it hardcoded in the Go code.
As I understand it can be painful to modify the cue schemas in a go code, we could keep it in a cue file, but separated from the rest of the schemas. And then we have a `generated` step that is injected the cue schema in the go code.

Reply:
@sjcobb I've changed it, the base defs are now retrieved from .cue files (located near the code).

Reply:
Chiming in late here.

We're going through a lot of rapid iteration on the individual panel schema (and soon possibly layout schema...), would it be feasible to set all the cue validation for individual panels to `any` (`_`) until they're more stable?

The perses project is still fairly early stage and I would say we need to be optimizing for speed of development over backwards compatibility/breaking changes.

Not sure how much perses is used internally at amadeus, but chronosphere hasn't launched anything with dashboards yet, so we consider large parts of the schema, especially plugin/panel schema fairly unstable as we build out new visualization, layout, template variable and other plugins and customizations. This will be a source of friction if we're not aligned on what stage of the project we're in and what parts of the schema is stable.

Related, anything that requires coordination between frontend code (especially plugin code!) and go backend code beyond an http API contract will noticeably slow down iteration cycles, so I think it will be beneficial if we can find a way to remove this coupling; at least during development time.

Reply:
I'm not sure exactly what you mean by 

> anything that requires coordination between frontend code (especially plugin code!) and go backend code beyond an http API contract


The whole point of putting the chart definition in Cuelang is to be able to align the frontend and the backend on the definition.
Basically if the frontend needs to update a chart definition in a PR, then the related cue files need to be aligned. The goal is not to modify the backend, but only the cue files.

I don't see how changing the cue files when the chart definition need to be changed / adapted will slow down the frontend development.

And also because there was a big gap between the backend and the frontend, we took a lot of time to realign everything and it's not yet finished. So on my side I'm not really ok to stop this effort.

Today we are not able to deploy Perses in Amadeus because of this gap

Reply:
but the fact the definition is in Cuelang is something that will unlikely change and can be considered as stable, because it is something that will help us to verify statically the chart definition without changing the backend code.

Reply:
Just talked to @LukeTillman and @eunicorn over zoom, I think there's a few things we're misunderstanding about how we're using cuelang and what parts of the schema we consider stable (and what that means), so we want to have a higher bandwidth conversation about it over zoom rather than in this random thread in a PR that's probably going to lead to more misunderstandings.

So let's talk in person. I'll ping you guys on element.

Reply:
Alright, then let's talk latter. In the meantime we can merge this PR. All issues raised here have been fixed